### DISCUSSION TITLE
Alpine x-ref cannot reference into a Livewire component

### DISCUSSION THREAD

**Discussion by GC-Max** [livewire/livewire#7568] [7. Bugs (archived)]
### Livewire version

v3.3.0

### Laravel version

v10.37.3

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

The below blade template should update the value within the `<livewire:input />` component when the `<select>` changes.

**welcome.blade.php**
```html
  <div x-data>
      <select x-on:change="$refs.name.value = $event.target.options[$event.target.selectedIndex].text">
          <option>This WONT work</option>
          <option>ABC</option>
      </select>

      <br><br>

      <livewire:input />
  </div>
```
**Livewire Input Component**
```
<div>
    <input type="text" x-ref="name" id="name"/>
</div>
```

You get the following error when changing the drop downs selected value.

```
Alpine Expression Error: Cannot set properties of undefined (setting 'value')

Expression: "$refs.name.value = $event.target.options[$event.target.selectedIndex].text"
```

To replicate this issue please find attached a link to an example repo:
https://github.com/GC-Max/livewire-issue
Run `composer install` 
Run `php artisan serve`
Navigate to the root within the browser to be able to test this issue.

### Repository or Wirebox URL

https://github.com/GC-Max/livewire-issue

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@GC-Max yeah that's correct, Alpine can't access refs inside other Alpine components, and as Livewire is now built on Alpine, each Livewire component is an Alpine component. Hence why that won't be working.

**Reply by cyrillkalita** [reply]
@joshhanley ok, but why not within the component itself?
component:
```html
<div x-data="{value:null}">
    <label x-bind:class="($refs.radio.value == value) ? 'bg-red-100' : ''" x-cloak>
        <input {{ $attributes->merge(['type'  => 'radio']) }} x-model="value" x-ref="radio">
    </label>
</div>
```

When i console.log the `$ref.radio.value` the value is functional, but upon livewire reload is undefined

**Reply by joffreypersia** [reply]
Hi @joshhanley, thank you for your answer. I understand it better, too. 
However, I struggled in the past to learn that, and I suggest that the AlpineJS doc mention this limitation more clearly. 

I don't know if you're part of the team, but I missed the nuance as a non-native English reader.
